src/main/java/org/codehaus/plexus/archiver/AbstractArchiver.java

```diff
@@ -1168,4 +1169,31 @@ public void setFilenameComparator( Comparator<String> filenameComparator )
     {
         return filenameComparator;
     }
+
+    @Override
+    public void configureReproducible( int sourceDateEpoch )
```

Comment:
I am not fully conviced by this for several reasons:

* Why introduce `sourceDateEpoch` when `setLastModifiedDate` does the same. The normalization can still be performed.
* You take the option away to provide alternative stable modes by default

I would take another approach:

`configureReproducible()` which checks for a LMD set and aligns to UTC, checks for a file comparator, checks for a file and directory mode set. We actually don't care which exact values have been set as long as they have been set. I am still convinced that the caller provides sane values for reproducibility.

Reply:
because there is a standard for Reproducible Builds that I prefer to use instead of reinventing the wheel: https://reproducible-builds.org/specs/source-date-epoch/

I don't understand what "checks for a LMD set and aligns to UTC" means, nor the 3 following lines: does it just mean that you prefer to replace "int sourceDateEpoch" with a "long lastModifiedDate"?

Reply:
That's just a write up, not even an RFC. I see no standard here, just a guideline.

The current benefit I see in `configureReproducible()` is just the timezone normalization. The rest can already be achieved with builtin options (strictly speaking). I don't like to introduce another last datetime field which serves the same purpose as `setLastModifiedDate()`, but with less precision.

Let me rephrase my counter approach:

```
validateReproducibility() {
if (lmd == null)
 throw new...
// else maybe align, but this should actually happen when the zip entry is written because if I call `getLMD` I get another value back compared to the one I have written.

if (comparator != null)
  throw new ...

if (forcedFileMode == -1)
  throw new...

if (forcedDirMode == -1)
  throw new...

}
```

Moreover, one just could add  `setTimeZone()` and pass `Etc/UTC` and everything will be properly aligned for your last modified date.

Is that a bit clearer?

Reply:
I don't see why your API "validates" anything: it doesn't seem to have the same objective as mine.
My API is to configure the archiver in a reproducible way, defining an order (that costs CPU, then is a choice that I would not do by default) and fixing Unix mode: you can configure the archiver by hand if you want, or you can also write another API that you prefer to create another API.
Regarding timezone, zip is the only format that does not understand timezones: tar does not have this limitation. That why zip implementation overrides default implementation of convertSourceDateEpochToDate(...)

Reply:
I think I need to change the code to show you what I mean. I think we are talking about the same, but you still don't understand me.

Reply:
+1